### PR TITLE
Adding new validation

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- Added party role type validation to the validator

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class GetValidPartyRoleRequestTest {
+
+    public static final String COURT_CLASSIFICATION = "A";
```
Comment (akroon3r) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequestTest.java:15]: do we want these hard coded?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import ca.bc.gov.open.jag.efilingapi.api.model.DocumentProperties;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GetValidPartyRoleRequest {
+
+    private List<DocumentProperties> documents = new ArrayList<>();
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequest.java:11]: isn't this `documentTypes`??

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    private String courtLevel;
+    private String courtClassification;
+
+    public List<DocumentProperties> getDocuments() {
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequest.java:15]: same here, `getDocumentTypes`

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return documents;
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/GetValidPartyRoleRequest.java:16]: `documentTypes`

**Review by akroon3r — APPROVED** [review]

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/validator/GenerateUrlRequestValidatorImpl.java [hunk 5]
```diff
@@ -22,10 +35,43 @@ public Notification validate(GenerateUrlRequest initialPackage) {
 
     }
 
+    /**
+     * Validates parties
+     *
+     * if court file number is empty then at least 1 party is required
+     *
+     * Parties types must be valid based on the document types submitted
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/validator/GenerateUrlRequestValidatorImpl.java:43]:
```suggestion
     * Party types must be valid based on the document types submitted
```

**Review by sdevalapurkar-bcgov — CHANGES_REQUESTED** [review]

**Comment by akroon3r** [comment]
I didn't see anything on the first or second pass. maybe @TayGov can also approve as well.

**Review by sdevalapurkar-bcgov — APPROVED** [review]
Thank you for this refactoring and addition of notification functionality 💯 
LGTM

**Review by TayGov — APPROVED** [review]

**Pull request merged**